tensorflow_lite_support/python/test/task/vision/image_classifier_test.py

```diff
@@ -128,6 +164,129 @@ def test_classify_model(self, model_file_type, max_results,
+    ]
+
+    # Builds test data.
+    expected_result_dict = self.build_test_data(expected_categories)
+
+    # Comparing results.
+    self.assertDeepAlmostEqual(image_result_dict, expected_result_dict, places=5)
+
+  def test_allowlist_option(self):
+    # Creates classifier.
+    model_file = _ExternalFile(file_name=self.model_path)
+
+    classifier = self.create_classifier_from_options(
+      model_file, class_name_allowlist=['cheeseburger', 'guacamole'])
+
+    # Loads image.
+    image = tensor_image.TensorImage.from_file(self.test_image_path)
+
+    # Classifies the input.
+    image_result = classifier.classify(image, bounding_box=None)
+    image_result_dict = json.loads(json_format.MessageToJson(image_result))
+
+    # Expected results.
+    expected_categories = [
+      {'index': 934, 'score': 0.7399742007255554, 'class_name': "cheeseburger"},
+      {'index': 925, 'score': 0.026928534731268883, 'class_name': "guacamole"}
+    ]
+
+    # Builds test data.
+    expected_result_dict = self.build_test_data(expected_categories)
+
+    # Comparing results.
+    self.assertDeepAlmostEqual(image_result_dict, expected_result_dict, places=5)
+
+  def test_denylist_option(self):
+    # Creates classifier.
+    model_file = _ExternalFile(file_name=self.model_path)
+
+    classifier = self.create_classifier_from_options(
+      model_file, score_threshold=0.01, class_name_denylist=['cheeseburger'])
+
+    # Loads image
+    image = tensor_image.TensorImage.from_file(self.test_image_path)
+
+    # Classifies the input.
+    image_result = classifier.classify(image, bounding_box=None)
+    image_result_dict = json.loads(json_format.MessageToJson(image_result))
+
+    # Expected results.
+    expected_categories = [
+      {'index': 925, 'score': 0.026928534731268883, 'class_name': "guacamole"},
+      {'index': 932, 'score': 0.025737214833498, 'class_name': "bagel"},
+      {'index': 963, 'score': 0.010005592368543148, 'class_name': "meat loaf"}
+    ]
+
+    # Builds test data.
+    expected_result_dict = self.build_test_data(expected_categories)
+
+    # Comparing results.
+    self.assertDeepAlmostEqual(image_result_dict, expected_result_dict, places=5)
+
+  def test_combined_allowlist_and_denylist(self):
+    # Fails with combined allowlist and denylist
+    with self.assertRaisesRegex(
+        Exception,
+        r"INVALID_ARGUMENT: `class_name_whitelist` and `class_name_blacklist` "
+        r"are mutually exclusive options. "
+        r"\[tflite::support::TfLiteSupportStatus='2'\]"):
+      base_options = _BaseOptions(model_file=_ExternalFile(file_name=self.model_path))
+      classification_options = classification_options_pb2.ClassificationOptions(
+        class_name_allowlist=['foo'], class_name_denylist=['bar'])
+      options = _ImageClassifierOptions(
+        base_options=base_options, classification_options=classification_options)
+      _ImageClassifier.create_from_options(options)
+
 
```

Comment:
please add tests for max_results and score_threshold. Similarly, these tests don't need to compare the model output with a predefined list, but assert if it's logically correct.

Reply:
@khanhlvg 
- I've now changed them to assert if they function logically correct instead of comparing results with the ground truth. Similarly added a case to test the correctness of the `max_results` option too. 
- Also tidied up the code by adding global constants to be used for each test case.